### DISCUSSION TITLE
Classical graph metrics supporting emeddings?

### DISCUSSION THREAD

**Discussion by geoHeil** [stellargraph/stellargraph#1877] [Q&A]
Similar but a bit different to https://github.com/stellargraph/stellargraph/discussions/1873 I wonder how useful classical graph metrics (i.e. PageRank, betweenness-centrality, ...) are for features on i.e. graphSage.

I.e. is it useful to calculate the classical metrics as features in a first step and use these as additional attributes for a graph during the embedding procedure?

**Comment by adocherty** [comment]
Hi, thanks for the question.

The answer is definitely, often if there are few native features for nodes it is standard practice to use classical graph structural information (such as triangle count, centrality, and many others) as features.  This can be in addition to native node features, or can be used in cases where there are no native features.

This has been demonstrated in academic papers, but I cannot remember which ones. If I can find a reference I will let you know.
